pandas/core/internals.py

```diff
@@ -5304,6 +5305,22 @@ def concatenate_block_managers(mgrs_indexers, axes, concat_axis, copy):
         elif is_uniform_join_units(join_units):
             b = join_units[0].block.concat_same_type(
                 [ju.block for ju in join_units], placement=placement)
+        elif is_sparse_join_units(join_units):
+            values = concatenate_join_units(join_units, concat_axis, copy=copy)
+
+            if len(values.shape) == 2:
+                values = values[0]
+            else:
+                assert len(values).shape == 1
```

Comment:
This never happens in my testing... so it's probably safe to pull out but it also would be odd if we got to this place.

Reply:
sorry what I mean to say is... len(values).shape is always either 1 or 2 in my testing. I'm not really sure what to do if shape == 3 or something and it was a true ndarray that was sparse.